Add new backport labels to dev docs (#137418)
elastic/kibana: dev_docs/contributing/how_we_use_github.mdx

- `release_node:plugin_api_changes`: Specifies a changes to the plugin API and adds the PR to the Plugin API changes page in the Developer Guide
- `release_note:skip`: Omits the PR from release notes

The following labels are related to backporting PRs:

- `auto-backport`: Automatically backport this PR (to the branches related to
   version labels) after it's merged. Requires adding desired target versions labels.

> - `backport:prev-minor`: Automatically backport to one branch/version back from main.

Reviewer: By mentioning the main branch, this seems to imply these labels can't be used if your PR is against other branches. Is that true? If not, a clearer wording might be something like this:

```suggestion
- `backport:prev-minor`: Automatically backport to one lower minor version.
```

> - `backport:prev-major`: Automatically backport to whichever branch contains the next patch version of the previous major release.

Reviewer: Since the branch names don't reflect patch versions, it correct to simplify as follows?:
```suggestion
- `backport:prev-major`: Automatically backport to the latest minor version of one lower major version.
```

> - `backport:all-open`: Automatically backport to all open branches/versions. This is functionally equivalent to backport:prev-major at the time of writing.

Reviewer: It is unclear to me what is meant by "open". We can (and sometimes do) backport docs changes very far back across many versions, but I assume this isn't the intention?:
```suggestion
- `backport:all-open`: Automatically backport to all open branches/versions. This is functionally equivalent to backport:prev-major at the time of writing.
```

- `backport:skip`: This PR does not require backporting.

> - `backport`: This PR was backported (added by CI).

Reviewer 2: ```suggestion
- `backported`: This PR was backported (added by CI).
```
